src/addressResolvers/farcaster.ts

```diff
@@ -0,0 +1,151 @@
+import { capture } from '@snapshot-labs/snapshot-sentry';
+import { graphQlCall, Address, Handle, FetchError, isSilencedError, isEvmAddress } from './utils';
+import { getAddress } from 'viem';
+
+export const NAME = 'Farcaster';
+const FNAMES_API_URL = 'https://fnames.farcaster.xyz/transfers?name=';
+const NEYNAR_API_URL = 'https://api.neynar.com/v2/farcaster/user/';
+const API_KEY = process.env.NEYNAR_API_KEY ?? '';
+
+interface UserDetails {
+  username: Handle;
+  verified_addresses: {
+    eth_addresses: Address[];
+    sol_addresses: string[];
+  };
+  pfp_url: string;
+}
+
+interface ApiResponse {
+  [address: string]: UserDetails[];
+}
+
+interface UserResult {
+  username?: Handle;
+  eth_addresses?: Address[];
+  sol_addresses?: string[];
+  pfp_url?: string;
+}
+
+export async function fetchData<T>(url: string): Promise<T> {
+  const headers = {
+    Accept: 'application/json',
+    api_key: API_KEY
+  };
+  const response = await fetch(url, { headers });
+  if (!response.ok) {
+    const error = new FetchError(`Failed to fetch data from the API. Status: ${response.status}`);
+    if (!isSilencedError(error)) {
+      capture(error, { tags: { issue: 'api_fetch_failure' } });
+    }
+    throw error;
+  }
+  return response.json() as Promise<T>;
+}
+
+
+export async function lookupAddresses(addresses: Address[]): Promise<{ [key: Handle]: Address }> {
+  const results: { [key: Handle]: Address } = {};
+
+  try {
+    const url = buildLookupUrl(addresses);
+    const userDetails = await fetchData<ApiResponse>(url);
+    processUserDetails(userDetails, results);
+  } catch (error) {
+    handleLookupError(error, addresses);
+  }
+
+  return results;
+}
+
+export function buildLookupUrl(addresses: Address[]): string {
+  const addressesQuery = addresses.join(',');
```

Comment:
Addresses need to be filtered, to check if they're EvmAddress. 

If result is an empty array, we should return early here, and skip the whole workflow, else we're sending unsupported/invalid addresses to the API